outlines/structure.md

```diff
@@ -0,0 +1,56 @@
+# Application structure and code style
+
+1. What's different between Meteor and other kinds of apps?
+    1. All JavaScript, enables code sharing
+    2. Structure is flexible
+    3. Currently everything is about magic file names
+    4. App scope vs. package scope vs. file scope variables
+2. Client vs. server vs. both
+3. Smooth path from prototype to production, three sizes of apps, S, M, L
+4. General stuff
+    1. Filenames - named the same as the export
+        1. Every file exports one thing
+    2. Avoid depending on load order via namespacing and Meteor.startup
+        1. If you depend on load order, your best bet is a package
+    3. Naming guidelines for everything
+        1. https://github.com/meteor/guide/issues/9#issuecomment-148800085
```

Comment:
> Yes, but then you loose UI goodies. :-) Then stuff becomes like 90s CGI forms. :-)

Not if the Method is separated into validation and body. Thus this conversation.

> Anyway, why you have to extend Method for this? This is all what we needed: https://github.com/peerlibrary/peerlibrary/blob/master/lib/errors.coffee

Yeah the format of the validation error is the subject of https://github.com/meteor/guide/issues/51